docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
+5. If some flag is not satisfied, continue the loop without adding foodItem to the result
+6. If the flags loop is over, the flags conditions were satisfied, so add the foodItem to the result
 
-## Non-Functional Requirements
+**Class Diagram** 
+![ClassDiagram](images/FindCommandClassDiagram.png)
 
-{Give non-functional requirements}
 
-## Glossary
+**Sequence Diagram**
+![ClassDiagram](images/FindCommandSequenceDiagram.png)
+
+### List feature
+The list command is implemented using a `ListCommand` class which utilizes `COMMAND_WORD`
+attributes of all other commands within the `commands` package.
 
-* *glossary item* - Definition
+The list command is implemented as follows:
+1. New `ListCommand` object created will call the method `execute` of `Command` class.
+2. The method `execute` decides what to append to the string `printToUser` according to `foodList`. 
+3. `foodList` is an object of type `FoodList` that is passed as argument to `execute`, it consists of the list of food 
+that has been added by user.
+4. `foodList` contains of `Food` that has undergoes the `toString` method.
+5. `execute` will check if `numberOfFood` is greater than 0, in which `numberOfFood` is obtained from the `FoodList` 
+function `getNumberOfFood`.
+6. If `numberOfFood` is greater than 0,  the foodList in String will be appended to `printToUser`. If it is not, then 
+the empty `foodList` will not be appended to `printToUser`.
+7. After appending the `foodList`, the method `execute` will return an object called `CommandResult` and pass          
+`printToUser` as its argument.
+8. `Duke` will then call `printResult` method from `CommandResult` which will print the food list for the user.
+
+**Class Diagram**
+![ClassDiagram](images/ListCommandClassDiagram.png)
+
+**Object Diagram**
+![ClassDiagram](images/ListCommandSequenceDiagram.png)
+
+### Help feature
+The help command is implemented using a `HelpCommand` class which utilizes `COMMAND_WORD` 
+attributes of all other commands within the `commands` package. 
+
+The help command is implemented as follows:
+1. New `HelpCommand` object is created by passing in a String containing arguments from `Parser`.
+2. The constructor `HelpCommand` will split the arguments based on the `--` regex and store them in an array of
+strings called `filters`.
+3. The method `execute` of `Command` class will then be called all the way from `Duke` with `filters` as its argument. 
+4. This method decides what to append to the string `printToUser` as specified by the `filters`. 
+5. After looping through all the `filters`, this method will return an object called `CommandResult` and pass
+`printToUser` as its argument.
+6. `Duke` will then call `printResult` method from `CommandResult` which will print the necessary message.
+
+**Class Diagram**
+![ClassDiagram](images/HelpCommandClassDiagram.png)
+
+
+**Sequence Diagram**
+![ClassDiagram](images/HelpCommandSequenceDiagram.png)
```

Comment:
Is `out.println()` a method of `CommandResult`?

Reply:
Should there be a return arrow for the constructor of  `Command`?

Reply:
Is there supposed to be a return arrow from the Parser to the main Duke class?